$OpenBSD: patch-Download_hs,v 1.5 2014/12/18 20:16:59 kili Exp $

port to ghc-7.8

fix version number in the User-Agent string

--- Download.hs.orig	Tue Nov  8 16:49:38 2011
+++ Download.hs	Thu Dec 18 20:53:14 2014
@@ -43,7 +43,7 @@ import System.Posix.Process
 import System.Posix.Types
 import System.Posix.IO
 import Data.Hash.MD5
-import Control.Exception(evaluate)
+import qualified Control.Exception as E
 
 data Result = Success | Failure
             deriving (Eq, Show, Read)
@@ -59,7 +59,7 @@ d = debugM "download"
 i = infoM "download"
 
 curl = "curl"
-curlopts = ["-A", "hpodder v1.0.0; Haskell; GHC", -- Set User-Agent
+curlopts = ["-A", "hpodder v1.1.6; Haskell; GHC", -- Set User-Agent
             "-s",               -- Silent mode
             "-S",               -- Still show error messages
             "-L",               -- Follow redirects
@@ -74,7 +74,8 @@ getCurlConfig =
     do ad <- getAppDir
        return $ ad ++ "/curlrc"
 
-getsize fp = catch (getFileStatus fp >>= (return . Just . fileSize))
+getsize fp = (E.catch :: IO a -> (IOError -> IO a) -> IO a)
+             (getFileStatus fp >>= (return . Just . fileSize))
              (\_ -> return Nothing)
 
 {- | Begin the download process on the given URL.
@@ -118,7 +119,7 @@ finishGetURL dltok ec =
        let r = case ec of
                   Exited ExitSuccess -> Success
                   Exited (ExitFailure i) -> Failure
-                  Terminated _ -> Failure
+                  Terminated _ _ -> Failure
                   Stopped _ -> Failure
        if r == Success
           then do d $ "curl returned successfully; new size is " ++
